dynamodb/dynamodb.sim.w

```diff
@@ -0,0 +1,214 @@
+            state.set("endpoint", "http://0.0.0.0:{port}");
+          }
+        },
+      );
+
+      return () => {
+        docker.kill();
+      };
+    });
+
+    // The host will be ready when the endpoint is set.
+    new cloud.Service(inflight () => {
+      util.waitUntil(() => {
+        return state.tryGet("endpoint")?;
+      });
+    }) as "Wait Until";
+
+    // std.Node.of(this).hidden = true;
+  }
+
+  pub static of(scope: std.IResource): Host {
+    let uid = "DynamodbHost-7JOQ92VWh6OavMXYpWx9O";
+    let root = std.Node.of(scope).root;
+    let rootNode = std.Node.of(root);
+    let host = unsafeCast(rootNode.tryFindChild(uid)) ?? new Host() as uid in root;
+    nodeof(host).hidden = true;
+    return host;
+  }
+}
+
+pub class Table_sim impl dynamodb_types.ITable {
+  host: Host;
+  tableName: str;
+
+  new(props: dynamodb_types.TableProps) {
+    this.host = Host.of(this);
+
+    let tableName = this.node.addr;
+    let state = new sim.State();
+    this.tableName = state.token("tableName");
+
+    new cloud.Service(inflight () => {
+      let client = Util.createClient({
+        endpoint: this.host.endpoint,
+        region: "local",
+        credentials: {
+          accessKeyId: "local",
+          secretAccessKey: "local",
+        },
+      });
+
+      let attributeDefinitions = MutArray<Json> [];
+      for attributeDefinition in props.attributes {
+        attributeDefinitions.push({
+          AttributeName: attributeDefinition.name,
+          AttributeType: attributeDefinition.type,
+        });
+      }
+
+      let keySchemas = MutArray<Json> [];
+      keySchemas.push({
+        AttributeName: props.hashKey,
+        KeyType: "HASH",
+      });
+      if let rangeKey = props.rangeKey {
+        keySchemas.push({
+          AttributeName: rangeKey,
+          KeyType: "RANGE",
+        });
+      }
+
+      util.waitUntil(() => {
+        try {
+          client.createTable({
+            TableName: tableName,
+            AttributeDefinitions: attributeDefinitions.copy(),
+            KeySchema: keySchemas.copy(),
+            BillingMode: "PAY_PER_REQUEST",
+            StreamSpecification: {
+              StreamEnabled: true,
+              StreamViewType: "NEW_AND_OLD_IMAGES",
+            },
+          });
+
+          if let timeToLiveAttribute = props.timeToLiveAttribute {
+            client.updateTimeToLive({
+              TableName: tableName,
+              TimeToLiveSpecification: {
+                AttributeName: timeToLiveAttribute,
+                Enabled: true,
+              },
+            });
+          }
+
+          state.set("tableName", tableName);
+          return true;
+        } catch error {
+          return false;
+        }
+      });
+    });
+  }
+
+  pub connection(): dynamodb_types.Connection {
+    return {
+      endpoint: this.host.endpoint,
+      tableName: this.tableName,
+    };
+  }
+
+  pub setStreamConsumer(handler: inflight (dynamodb_types.StreamRecord): void) {
```

Comment:
Need some shutdown code here

Reply:
I haven't encountered a situation where this keeps running. If the table container stops being available, it will end up halting the loop.